mp/src/game/client/momentum/fx_mom_explosions.cpp

```diff
@@ -0,0 +1,175 @@
+    char *pszSound = "BaseExplosionEffect.SoundTF2";
+
+    if (pWeaponInfo)
+    {
+        // Explosions.
+        if (bIsWater)
+        {
+            if (Q_strlen(pWeaponInfo->m_szExplosionWaterEffect) > 0)
+                pszEffect = pWeaponInfo->m_szExplosionWaterEffect;
+        }
+        else
+        {
+            if (bIsPlayer || bInAir)
+            {
+                if (Q_strlen(pWeaponInfo->m_szExplosionPlayerEffect) > 0)
+                    pszEffect = pWeaponInfo->m_szExplosionPlayerEffect;
+            }
+            else
+            {
+                if (Q_strlen(pWeaponInfo->m_szExplosionEffect) > 0)
+                    pszEffect = pWeaponInfo->m_szExplosionEffect;
+            }
+        }
+
+        // Sound.
+        if (Q_strlen(pWeaponInfo->m_szExplosionSound) > 0)
+        {
+            pszSound = pWeaponInfo->m_szExplosionSound;
+        }
+    }
+
+    CLocalPlayerFilter filter;
+    if (mom_rj_sounds.GetInt() == 0)
+    {
+        //return;
+    }
+    if (mom_rj_sounds.GetInt() == 1)
+    {
+        char *pMomSound = "BaseExplosionEffect.Sound";
+        C_BaseEntity::EmitSound(filter, SOUND_FROM_WORLD, pMomSound, &vecOrigin);
+    }
+    if (mom_rj_sounds.GetInt() == 2)
+    {
+        C_BaseEntity::EmitSound(filter, SOUND_FROM_WORLD, pszSound, &vecOrigin);
+    }
+
+    if (mom_rj_particles.GetInt() == 0)
+    {
+    }
+    else if (mom_rj_particles.GetInt() == 1)
+    {
+    }
+    else if (mom_rj_particles.GetInt() == 2)
+    {
+        DispatchParticleEffect(pszEffect, vecOrigin, angExplosion);
+    }
+}
+
+class C_TETFExplosion : public C_BaseTempEntity
+{
+  public:
+    DECLARE_CLASS(C_TETFExplosion, C_BaseTempEntity);
+    DECLARE_CLIENTCLASS();
+
+    C_TETFExplosion(void);
+
+    virtual void PostDataUpdate(DataUpdateType_t updateType);
+
+  public:
+    Vector m_vecOrigin;
+    Vector m_vecNormal;
+    CWeaponID m_iWeaponID;
+    ClientEntityHandle_t m_hEntity;
+};
+
+C_TETFExplosion::C_TETFExplosion(void)
+{
+    m_vecOrigin.Init();
+    m_vecNormal.Init();
+    m_iWeaponID = WEAPON_NONE;
+    m_hEntity = INVALID_EHANDLE_INDEX;
+}
+
+void C_TETFExplosion::PostDataUpdate(DataUpdateType_t updateType)
+{
+    VPROF("C_TETFExplosion::PostDataUpdate");
+
+    TFExplosionCallback(m_vecOrigin, m_vecNormal, m_iWeaponID, m_hEntity);
+}
+
+static void RecvProxy_ExplosionEntIndex(const CRecvProxyData *pData, void *pStruct, void *pOut)
+{
+    int nEntIndex = pData->m_Value.m_Int;
+    ((C_TETFExplosion *)pStruct)->m_hEntity =
+        (nEntIndex < 0) ? INVALID_EHANDLE_INDEX : ClientEntityList().EntIndexToHandle(nEntIndex);
+}
+
+IMPLEMENT_CLIENTCLASS_EVENT_DT(C_TETFExplosion, DT_TETFExplosion, CTETFExplosion)
+RecvPropFloat(RECVINFO(m_vecOrigin[0])), RecvPropFloat(RECVINFO(m_vecOrigin[1])),
+    RecvPropFloat(RECVINFO(m_vecOrigin[2])), RecvPropVector(RECVINFO(m_vecNormal)), RecvPropInt(RECVINFO(m_iWeaponID)),
+    RecvPropInt("entindex", 0, SIZEOF_IGNORE, 0, RecvProxy_ExplosionEntIndex), END_RECV_TABLE()
```

Comment:
Every property should be on its own line.
```
IMPLEMENT_CLIENTCLASS_EVENT_DT(C_TETFExplosion, DT_TETFExplosion, CTETFExplosion)
    RecvPropFloat(RECVINFO(m_vecOrigin[0])),
    RecvPropFloat(RECVINFO(m_vecOrigin[1])),
    RecvPropFloat(RECVINFO(m_vecOrigin[2])),
    RecvPropVector(RECVINFO(m_vecNormal)), 
    RecvPropInt(RECVINFO(m_iWeaponID)),
    RecvPropInt("entindex", 0, SIZEOF_IGNORE, 0, RecvProxy_ExplosionEntIndex), 
END_RECV_TABLE()
```

Secondly, why use `RecvPropFloat` when you already are using `RecvPropVector` right below it?
```
    RecvPropFloat(RECVINFO(m_vecOrigin[0])),
    RecvPropFloat(RECVINFO(m_vecOrigin[1])),
    RecvPropFloat(RECVINFO(m_vecOrigin[2])),
```
=>
```
    RecvPropVector(RECVINFO(m_vecOrigin)),
```

Reply:
When using `RecvPropVector`, I get the following error in the console when trying to load a map:
`Missing RecvProp for DT_TETFExplosion - DT_TETFExplosion/m_vecOrigin[0]`

Reply:
Nevermind, I think it's just because I forgot to set it in the serverclass datatable as well.